src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -50,6 +50,12 @@ public static string HtmlEncode(string value)
             return StringBuilderCache.GetStringAndRelease(sb);
         }
 
+        public static void HtmlEncode(string value, TextWriter output)
+        {
+            // Since we already have a private method that decodes the value using a StringBuilder we don't want to have duplicate code in order to have a better mantainance.
+            output.Write(HtmlEncode(value));
```

Comment:
@justinvp Your solution seems like it would be slower for the main path, since an extra function call would be added for every char/string appended. Since it's more common to call the overload that returns a string rather than one that writes it to a `TextWriter`, it seems like what's currently checked in is the best solution.